Allow adding only-quirk instance IDs from quirk files

hughsie opened the pull request:
We never want to match firmware to these GUIDs.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

superm1 reviewed (commented):
This needs new documentation as it's technically a new key.

hughsie commented:
> This needs new documentation

In the plugin tutorial?

superm1 commented:
> > This needs new documentation
> 
> In the plugin tutorial?

I think Gtkdoc for the "Quirks" key needs a new key for this

hughsie commented:
> I think Gtkdoc for the "Quirks" key needs a new key for this

I forgot all about src/README.md -- I've moved it to docs and added it to the generated HTML.

superm1 commented on libfwupdplugin/fu-device.c:
```diff
@@ -1871,6 +1871,12 @@ fu_device_set_quirk_kv(FuDevice *self, const gchar *key, const gchar *value, GEr
 			fu_device_add_guid(self, sections[i]);
 		return TRUE;
 	}
+	if (g_strcmp0(key, FU_QUIRKS_GUID "[quirk]") == 0) {
```
I think this should be it's own `#define`, otherwise it's going to be harder to find in the codebase when looking for `Guid[quirk]`

hughsie replied:
Fixed; I've move the docs commit to a future PR -- I think we need to merge the src/README.md into the existing fu-quirks.h docs -- it's weird having 90% the same in two places.
